src/libxtp/gwbse/gwbse.cc

```diff
@@ -703,18 +711,39 @@ bool GWBSE::Evaluate() {
 
     BSE bse = BSE(*_pLog, Mmn);
     bse.configure(_bseopt, _orbitals.RPAInputEnergies(), Hqp);
+
+    // store the direct contribution to the static BSE results
+    Eigen::VectorXd Hd_static_contrib_triplet;
+    Eigen::VectorXd Hd_static_contrib_singlet;
+
     if (_do_bse_triplets) {
       bse.Solve_triplets(_orbitals);
       XTP_LOG(Log::error, *_pLog)
           << TimeStamp() << " Solved BSE for triplets " << flush;
-      bse.Analyze_triplets(_fragments, _orbitals);
+      Hd_static_contrib_triplet = bse.Analyze_triplets(_fragments, _orbitals);
     }
 
     if (_do_bse_singlets) {
       bse.Solve_singlets(_orbitals);
       XTP_LOG(Log::error, *_pLog)
           << TimeStamp() << " Solved BSE for singlets " << flush;
-      bse.Analyze_singlets(_fragments, _orbitals);
+      Hd_static_contrib_singlet = bse.Analyze_singlets(_fragments, _orbitals);
```

Comment:
See the comment above, I would pay the price for the one more iteration to make the code much cleaner and not have this coupling here.

Reply:
Yeah, I had the static case inside the Perturbative function before, but it means that we have to recalc the static RPA. But indeed, probably no much extra effort compared to the rest of the iterations. Will go back to that one.

Reply:
Changed this one.